File: sdk/dotnet-guest/examples/random.cs
using Lsf.Guest;
using Raw = ServiceWorld.wit.Imports.latent.random.IRandomImports;
namespace ServiceWorld.wit.Exports.tests.random;
public class ApiExportsImpl : IApiExports
{
    public static ulong Run(uint which, string text, ulong handle)
    {
        if (which == 1) { _ = Random.U64().AsOk; return 8; }
        var result = Random.Bytes(which == 2 ? uint.MaxValue : 32);
        if (!result.IsOk && result.AsErr.Tag == Raw.RandomError.Tags.InvalidLength) return 10;
        return (ulong)result.AsOk.Length;
    }
}

File: sdk/typescript-guest/capabilities/random.ts
import * as raw from 'latent:random/random@0.1.0';
import { call } from './result.js';
export type { RandomError } from 'latent:random/random@0.1.0';
const errors: readonly raw.RandomError['tag'][] = ['invalid-length', 'budget-exhausted', 'unavailable'];
export function bytes(length: number) { return call<Uint8Array, raw.RandomError>(() => raw.bytes(length), errors); }
export function u64() { return call<bigint, raw.RandomError>(() => raw.u64Value(), errors); }
